### PR TITLE
Refs task #391647 [CR] Validate condition fields by a given payload params

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets

- [#391647](https://dev.sun-asterisk.com/issues/391647)

## What's this PR do ?

- [x] Validate condition fields with payload params
- [x] Unit test for validate condition fields with payload params

## Library
*(List package, library third party add new include version)*

- N/A

## Performance

- [ ] Resolved n + 1 query
- [ ] Run explain query already
- [ ] Time run rake task : 1000 ms

## Checklist

- [x] It was tested in local success?
- [x] Fill link PR into ticket and the opposite
- [x] Note purpose/reason, solution, scope of influence into ticket

## ENV note

```
```

## Notes
*(Other notes)*
image

image

### CODE REVIEW

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr></table>

File: app/Rules/MatchPayloadParams.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+     * Determine if the validation rule passes.
+     *
+     * @param  string  $attribute
+     * @param  mixed  $value
+     * @return bool
+     */
+    public function passes($attribute, $value)
+    {
+        $payload = json_decode($this->payloadParams);
+        for ($i = 0; $i < count($value); $i++) {
```
Comment (framgiaciv3) [inline app/Rules/MatchPayloadParams.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Rules/MatchPayloadParams.php</br>**Line**: 35</br>String "return " does not require double quotes; use single quotes instead

Comment (framgiaciv3) [inline app/Rules/MatchPayloadParams.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Rules/MatchPayloadParams.php</br>**Line**: 35</br>String ";" does not require double quotes; use single quotes instead

File: app/Rules/MatchPayloadParams.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            try {
+                eval("return " . $value[$i] . ";");
```
Comment (framgiaciv3) [inline app/Rules/MatchPayloadParams.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Rules/MatchPayloadParams.php</br>**Line**: 37</br>String "field" does not require double quotes; use single quotes instead

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr></table>

File: app/Rules/MatchPayloadParams.php (hunk 1 above)
Comment (framgiaciv3) [inline app/Rules/MatchPayloadParams.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Rules/MatchPayloadParams.php</br>**Line**: 35</br>String "return " does not require double quotes; use single quotes instead

Comment (framgiaciv3) [inline app/Rules/MatchPayloadParams.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Rules/MatchPayloadParams.php</br>**Line**: 35</br>String ";" does not require double quotes; use single quotes instead

File: app/Rules/MatchPayloadParams.php (hunk 2 above)
Comment (framgiaciv3) [inline app/Rules/MatchPayloadParams.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Rules/MatchPayloadParams.php</br>**Line**: 37</br>String "field" does not require double quotes; use single quotes instead

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr></table>

**Review by quanghung97 — APPROVED** [review]
ok

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr></table>

**Review by nguyenthinhhn — APPROVED** [review]

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr></table>

**Comment by phucqd-1305** [comment]
OK

**Pull request merged**